client/components/edit/ObjectLoader.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { useEffect } from "react";
+import { useEditorContext } from "../../context/EditorContext";
+import CircularProgress from "@mui/material/CircularProgress";
+
+export interface ObjectLoaderProps {
+    pid: string;
+}
+
+export const ObjectLoader = ({ pid }: ObjectLoaderProps): React.ReactElement => {
+    const {
+        state: { objectDetailsStorage },
+        action: { loadObjectDetailsIntoStorage },
+    } = useEditorContext();
+    const loaded = Object.prototype.hasOwnProperty.call(objectDetailsStorage, pid);
+
+    useEffect(() => {
+        if (!loaded) {
+            loadObjectDetailsIntoStorage(pid);
+        }
+    }, [loaded]);
+    return !loaded ? (
+        <>
+            &nbsp;
+            <CircularProgress size="1em" />
+        </>
+    ) : (
```

Comment:
you can just return null for the else statement.

Reply:
Thanks! Improved here: https://github.com/FalveyLibraryTechnology/VuDLPrepJS/pull/147/commits/f7ecded87ea7bcbf6df3cd2d4ed37c91e36aecdd